### PR TITLE
 Musl: add more stdlib definitions and enable HaveMemstream

### PR DESCRIPTION
**Pull request by Cogitri**

### CODE REVIEW

File: src/core/sys/posix/stdlib.d [hunk 1]
```diff
@@ -575,9 +575,45 @@ else version (CRuntime_Bionic)
...
+    void   setkey(const scope char*);
+    char*  setstate(const scope char*);
+    void   srand48(c_long);
+    void   srandom(uint);
+    int    unlockpt(int);
+
+  static if ( __USE_LARGEFILE64 )
+  {
+    int    mkstemp64(char*);
+    alias  mkstemp64 mkstemp;
```
Comment (Geod24) [inline src/core/sys/posix/stdlib.d:611]: This `alias` means that any call to `mkstemp` will link in `mkstemp64`. That's uncommon in the C bindings and rarely do what you want. Is that intended (e.g. `mkstemp` is a define) ?

File: src/core/sys/posix/stdio.d [hunk 2]
```diff
@@ -332,6 +332,8 @@ else version (OpenBSD)                      // as of OpenBSD 5.4
     version = HaveMemstream;
 else version (CRuntime_UClibc)
     version = HaveMemstream;
+else version(CRuntime_Musl)
```
Comment (Geod24) [inline src/core/sys/posix/stdio.d:335]:
```suggestion
else version (CRuntime_Musl)
```

File: src/core/sys/posix/stdlib.d (hunk 1 above)
Comment (Cogitri) [reply src/core/sys/posix/stdlib.d:611]: Yup, it's a define:

```
#if defined(_LARGEFILE64_SOURCE) || defined(_GNU_SOURCE)
#define mkstemp64 mkstemp
#define mkostemp64 mkostemp
#if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
#define mkstemps64 mkstemps
#define mkostemps64 mkostemps
#endif
#endif
```

File: src/core/sys/posix/stdio.d (hunk 2 above)
Comment (Cogitri) [reply src/core/sys/posix/stdio.d:335]: Ah thanks. Is there some standard formatter for the druntime? dfmt didn't like the alignment by variable name in this file.

Comment (Geod24) [reply src/core/sys/posix/stdio.d]: `make -f posix.mak style`

Comment (Cogitri) [reply src/core/sys/posix/stdio.d]: Will do that next time, thanks :)

File: src/core/sys/posix/stdio.d [hunk 3]
```diff
@@ -278,6 +309,30 @@ else version (CRuntime_UClibc)
     off_t ftello(FILE*);
   }
 }
+else version(CRuntime_Musl)
```
Comment (Geod24) [inline src/core/sys/posix/stdio.d:312]:
```suggestion
else version (CRuntime_Musl)
```

That's why the CI is red BTW

File: src/core/sys/posix/stdio.d [hunk 4]
```diff
@@ -332,6 +387,8 @@ else version (OpenBSD)                      // as of OpenBSD 5.4
     version = HaveMemstream;
 else version (CRuntime_UClibc)
     version = HaveMemstream;
+else version (CRuntime_Musl)
+    version = HaveMemstream;                // http://git.musl-libc.org/cgit/musl/commit/src/stdio/open_memstream.c?id=b158b32a44d56ef20407d4285b58180447ffff1f
```
Comment (Geod24) [inline src/core/sys/posix/stdio.d:391]:
```suggestion
// http://git.musl-libc.org/cgit/musl/commit/src/stdio/open_memstream.c?id=b158b32a44d56ef20407d4285b58180447ffff1f
else version (CRuntime_Musl)
    version = HaveMemstream;
```

The style checker also complains about lines > 120 chars

**Pull request merged**